src/foam/dao/README.md

```diff
@@ -0,0 +1,174 @@
+
+```java
+new foam.dao.EasyDAO.Builder(x)
+  .setOf(MODEL_PATH.getOwnClassInfo())
+  .setAuthorize(false)
+  .setRuler(false)
+  .setDecorator(new faom.dao.DAO1(new foam.dao.DAO2(null)))
+  .setInnerDAO(new INDAO())
+  .build();
+```
+
+This will create the following decorator chain,
+
+```
+EasyDAO -> DAO1 -> DAO2 -> INDAO
+```
+
+&nbsp;
+
+#### Debugging Tips and Tricks
+
+Unfortunately, setting up a decorator chain can be a very buggy and frustrating experience. The best way to set up a chain is sequentially, testing its function at each step of the way. This is because any bug that might be caused by a single step of the process can be very difficult to pinpoint once the entire decorator chain is set up. A sequential approach will give you a strong indication to what is causing each bug as it pops up. The best way to use this approach is to disable any decorators that may be added by default at first, and slowly add them one by one, taking care to prioritize adding first decorators that may conflict with others such as those performing authorization checks.
+
+It will help to visualize what decorator chain the EasyDAO is creating at each step of the way to help identify possible culprits and mismatched orders of DAOs. This can be done using the following piece of logic to walk the chain of decorators and print them out one by one,
+
+```java
+  foam.dao.DAO delegate = EASYDAO;
+
+  while ( delegate instanceof foam.dao.ProxyDAO) {
+    System.out.println(delegate.getClass().getSimpleName());
+    delegate = ((foam.dao.ProxyDAO) delegate).getDelegate();
+  }
+
+  System.out.println(delegate.getClass().getSimpleName());
+```
+
+This in fact comes built in to each EasyDAO in a method called printDecorators which you may wish to use. One of the advantages of having an EasyDAO decorator at the tail of each chain is simply being able to call it using the built dao,
+
+```java
+EasyDAO dao = new foam.dao.EasyDAO.Builder(x)
+  ...
+  ...
+  ...
+  .build();
+
+  dao.printDecorators();
+```
+
+&nbsp;
+&nbsp;
+
+## Customizing EasyDAO for the Client Side
+
+#### Overwriting Properties
+
+Customizing EasyDAO on the client side is a bit dificult do to the nature of how it is implemented. It is however, not impossible. The first step to take would be to create a foam model extending the original EasyDAO. Here you can override properties and even add some of your own. Since all the logic of creating the decorator chain is placed within a single property it is a bit hard to modify. However, EasyDAO comes equipped with a method allowing the user to specify their own unique decorator chain using the properties that they have added. This is known as getOuterDAO. 
```

Comment:
s/dificult/difficult/
s/daos/DAOs/

What does this mean: "This is a simple way to EasyDAO whichever DAOs you like"?
Add , after "For instance".
s/purely EasyDAO/purely with EasyDAO/